src/plugins_3rdparty/primer3/src/Primer3Plugin.h

```diff
@@ -31,6 +32,12 @@
 
 namespace U2 {
 
+/*
+ * Primer3Task should lock this resource before run and unlock after finish
+ * It's required because the original "primer3" tool doesn't support parallel calculations
+ */
+#define PRIMER3_STATIC_LOCK 6
```

Comment:
Give it a suffix "_RESOURCE" and use some more or less big random  number so there is no chance of overlap with other locks

Reply:
OK